### PR TITLE
OHRM5X-1617: Fix recruitment - candidate history permission related issues

### PR DESCRIPTION
**Pull request by SandeepaOHRM**

### CODE REVIEW

File: src/plugins/orangehrmRecruitmentPlugin/Api/HiringManagerAPI.php [hunk 1]
```diff
@@ -26,26 +26,43 @@
...
+        $vacancies = $this->getVacancyService()
+            ->getVacancyDao()
+            ->getVacancyListGroupByHiringManager($vacancySearchFilterParams);
+        $hiringManagerEmpNumbers = array_map(function ($vacancy) {
+            return $vacancy->getHiringManager()->getEmpNumber();
         }, $vacancies);
+        $employeeSearchFilterParams = new EmployeeSearchFilterParams();
+        $this->setSortingAndPaginationParams($employeeSearchFilterParams);
+        $employeeSearchFilterParams->setEmployeeNumbers($hiringManagerEmpNumbers);
+        $hiringMangers = $this->getEmployeeService()->getEmployeeList($employeeSearchFilterParams);
```
Comment (RajithaKumara) [inline src/plugins/orangehrmRecruitmentPlugin/Api/HiringManagerAPI.php:60]:
```suggestion
        $hiringManagers = $this->getEmployeeService()->getEmployeeList($employeeSearchFilterParams);
```
typo

File: src/plugins/orangehrmRecruitmentPlugin/Controller/WorkflowActionHistoryController.php [hunk 2]
```diff
@@ -23,18 +23,23 @@
...
 use OrangeHRM\Entity\CandidateHistory;
+use OrangeHRM\Entity\Interview;
 use OrangeHRM\Framework\Http\Request;
 use OrangeHRM\Recruitment\Traits\Service\CandidateServiceTrait;
 
 class WorkflowActionHistoryController extends AbstractVueController implements CapableViewController
 {
     use UserRoleManagerTrait;
     use CandidateServiceTrait;
+    use VueComponentPermissionTrait;
```
Comment (RajithaKumara) [inline src/plugins/orangehrmRecruitmentPlugin/Controller/WorkflowActionHistoryController.php:41]:
```suggestion
```
seems no need

File: src/plugins/orangehrmRecruitmentPlugin/Dto/CandidateActionHistory.php [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+class CandidateActionHistory
+{
+    /**
+     * @return int[]
+     */
+    public function getAccessibleCandidateActionHistoryIds(): array
+    {
+        return array_keys(
+            array_replace(
```
Comment (RajithaKumara) [inline src/plugins/orangehrmRecruitmentPlugin/Dto/CandidateActionHistory.php:33]: why we need `array_replace` :thinking: ?

**Review by RajithaKumara — CHANGES_REQUESTED** [review]

File: src/plugins/orangehrmRecruitmentPlugin/Controller/WorkflowActionHistoryController.php (hunk 2 above)
Comment (SandeepaOHRM) [reply src/plugins/orangehrmRecruitmentPlugin/Controller/WorkflowActionHistoryController.php:41]: done

File: src/plugins/orangehrmRecruitmentPlugin/Api/HiringManagerAPI.php (hunk 1 above)
Comment (SandeepaOHRM) [reply src/plugins/orangehrmRecruitmentPlugin/Api/HiringManagerAPI.php:60]: fixed

**Review by RajithaKumara — APPROVED** [review]

**Pull request merged**